src/plotting/util.tsx

```diff
@@ -0,0 +1,13 @@
+import { SelectedPlotChannel } from '../app.types';
+
+// TODO test this
+export const deepCopySelectedPlotChannels = (
```

Comment:
If you want to actually deep copy - you're best using `JSON.stringify` and `JSON.parse` - this method works fine as long as the options in `options` are primitives and not objects themselves (as otherwise those underlying objects are still the same by reference)

Reply:
That was an alternative I considered. Happy to make the change

Reply:
It's probably best to make it to prevent any future accident problems by the fact that this is only a single level deep copy. I had similar problems with the 2 y axis e2e test where I was accidentally modifying the underlying mock request data instead of deep copying because I just did object spread instead of JSON stringify&parse

Reply:
Resolved in https://github.com/ral-facilities/operationsgateway/pull/102/commits/19fbd764a1ca526d887c958298bcb5adcc5454c7